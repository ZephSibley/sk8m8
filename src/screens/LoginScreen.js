import React from 'react';
import {Link} from 'react-router-dom';
import Button from '@material-ui/core/Button';

import '../styles/screens.css'
import formStyles from '../styles/forms';
import ValidInput from '../components/ui/ValidInput';

const LoginScreen = () => {
  const formClasses = formStyles()

  const loginHandler = async () => {
    //await AsyncStorage.setItem('userToken', 'abc');
    console.log('login')
  }

  const signupHandler = () => {
    console.log('signup')
  }

  return (
    <div className={'screen'}>
      <h1>S K 8 M 8</h1>
      <form className={formClasses.form}>
        <ValidInput
          label='username'
          validationrules={{
            minLength: 3,
          }}
        />
        <ValidInput
          label='password'
          validationrules={{
            minLength: 6,
          }}
        />
        <input
          style={{ display: 'none' }}
          id="button-submit"
          multiple
          type="submit"
        />
        <label htmlFor="button-submit">
          <Button
            variant="contained"
            color="primary"
            component="span"
            className={formClasses.submit}
            onClick={() => loginHandler()}
          >
            Log in
          </Button>
        </label>
      </form>
      <Link 
        to='/signup'
        style={{ textDecoration: 'none' }}
      >
        <Button
          variant="outlined"
          color="primary"
          component="span"
        >
          Sign up
        </Button>
      </Link>
    </div>
  );
};

export default LoginScreen;