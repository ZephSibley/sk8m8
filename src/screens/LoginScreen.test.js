import React from 'react';
import renderer from 'react-test-renderer';
import LoginScreen from './LoginScreen';

test('Basic Rendering', () => {
    const component = renderer.create(
        <LoginScreen />
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot();
})