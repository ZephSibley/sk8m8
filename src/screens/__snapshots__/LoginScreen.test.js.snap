// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic Rendering 1`] = `
<div
  className="form-screen"
>
  <video
    autoPlay={true}
    className="video-background"
    muted={true}
  >
    <source
      src="front_video_3.mp4"
      type="video/mp4"
    />
  </video>
  <h1>
    S K 8 M 8
  </h1>
  <div>
    <form
      className="makeStyles-form-1"
      onSubmit={[Function]}
    >
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
          data-shrink={false}
        >
          email
        </label>
        <div
          className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            className="MuiInputBase-input MuiInput-input"
            disabled={false}
            name="email"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="email"
            value=""
          />
        </div>
        
      </div>
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
          data-shrink={false}
        >
          password
        </label>
        <div
          className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            className="MuiInputBase-input MuiInput-input"
            disabled={false}
            name="password"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="password"
            value=""
          />
        </div>
        
      </div>
      <span
        className="MuiFormHelperText-root Mui-error"
      >
        
      </span>
      <button
        className="MuiButtonBase-root MuiButton-root makeStyles-submit-5 MuiButton-contained MuiButton-containedPrimary Mui-disabled Mui-disabled"
        disabled={true}
        margin="normal"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={-1}
        type="submit"
      >
        <span
          className="MuiButton-label"
        >
          Log in
        </span>
      </button>
    </form>
  </div>
  <p>
    No account?
  </p>
  <a
    href="/signup"
    onClick={[Function]}
    style={
      Object {
        "textDecoration": "none",
      }
    }
  >
    <span
      aria-disabled={false}
      className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        className="MuiButton-label"
      >
        Sign up
      </span>
    </span>
  </a>
</div>
`;
